### ISSUE TITLE
Splash screen should pop before daily update check

### ISSUE THREAD

**Issue by hinerm** [fiji/fiji#366]
The update check is terribly slow. I actually thought my Fiji was just broken this morning because it took so long to pop up. At the very least, having an indication that something is happening would be very helpful.

**Comment by ctrueden** [comment]
Fiji Future, right? Splash screen is definitely first. Happens before any classes besides app-launcher ones are even loaded, much less executed. Something else weird is going on...

**Comment by hinerm** [comment]
Yes this is just in Fiji Future. In Fiji classic the splash screen pops before the update check.

It is definitely related to the update check because it only happens the first time I launch.

See also https://github.com/fiji/fiji/issues/367 re: the increased length of time this check now takes, which compounds the confusion

**Comment by hinerm** [comment]
It is 100% impossible to pop a splash screen before the delay, because the delay is taking place while initializing the first Java graphical component.

I guess if the splash screen was generated by native code it would work but let's not.

**Comment by ctrueden** [comment]
> I guess if the splash screen was generated by native code it would work but let's not.

It should be really easy, though! Java has a built-in `-splash` flag for loading and displaying a splash screen natively. (It also have a Java-side [java.awt.SplashScreen](https://docs.oracle.com/javase/8/docs/api/java/awt/SplashScreen.html)... I cannot remember why I wrote my own `Splash` class in app-launcher, instead of using it. :confused:)
